src/components/spreadsheet/use-spreadsheet-equipments.ts

```diff
@@ -144,7 +159,80 @@ export const useSpreadsheetEquipments = (
                     setIsFetching(false);
                 });
         }
-    }, [equipment, shouldFetchEquipments, studyUuid, currentNode?.id, dispatch, formatFetchedEquipments]);
+    }, [
+        equipment,
+        shouldFetchEquipments,
+        studyUuid,
+        currentNode?.id,
+        dispatch,
+        formatFetchedEquipments,
+        customColumnsDefinitions,
+    ]);
+
+    useEffect(() => {
+        if (studyUuid && equipments != null && Object.keys(customColumnsDefinitions).length > 0) {
+            const aliases = customColumnsNodesAliases.map((nodeAlias) => nodeAlias.alias);
+            const aliasesAlreadyFound: string[] = [];
+            const currentTabCustomColumnsDef = customColumnsDefinitions[tablesNames[tabIndex]];
+            const spreadsheetEquipmentType = toSpreadsheetEquipmentType(tablesNames[tabIndex]);
+            const customColumns = currentTabCustomColumnsDef.columns;
+            let fetchers: Promise<any>[] = [];
+            let additionalEquipmentsByNodes: Record<string, Record<SpreadsheetEquipmentType, Identifiable[]>> = {};
+            if (spreadsheetEquipmentType) {
+                customColumns.forEach((column) => {
+                    const pattern: string = aliases.map((alias) => `\\b${alias}\\b\\.`).join('|'); // pattern to find the aliases in the expression
+                    const regex = new RegExp(pattern, 'g');
+                    const aliasesFoundInFormula = column.formula?.match(regex);
+                    aliasesFoundInFormula?.forEach((aliasWithDot) => {
+                        const alias = aliasWithDot.slice(0, -1); //because we searched for "alias." with the regex
+                        //To avoid fetching multiple time the same data because it's used in multiple column
+                        if (!aliasesAlreadyFound.includes(alias)) {
+                            aliasesAlreadyFound.push(alias);
```

Comment:
Why do you introduce all this complexity ? Why not just fetching all the data related to all the nodes defined in the customColumnsNodesAliases ?

Reply:
but that's precisely what i'm trying to do here or i'm not understanding what you're saying ? I can't just fetch the data for every customcolumn without keeping trace of what i already fetch or i would be doing 50 time the same fetch

Reply:
Oh ok i got it wrong, I understand what you mean, but it's not great to do that, we would be fetching a lot of data we don't actually need if we fetch all the data of the nodes that are found in aliases

Reply:
And I'm not a big fan of parsing formulas like that I think for now we should avoid this.

Reply:
And for me you only have one fetch to do when you add a new node in the node aliases list, this is absolutely not related to custom columns or even "traditional" columns. It's just that you want to load additionnal data

Reply:
I don't know... So the user add 2 customs columns one "alias1.maxP" and "alias2.maxP" only for the generators and we would end up with 3 whole network loaded in the front. That seems so unreasonable, that's why i did it this way.

Reply:
> And for me you only have one fetch to do when you add a new node in the node aliases list, this is absolutely not related to custom columns or even "traditional" columns. It's just that you want to load additionnal data

But it is related, this node alias option is only here to define the aliases used in... the custom columns. It was not defined as a load the network with this id. It's a this alias = this node in the custom column

Reply:
For now, but in a really near future all columns will be "custom" columns, then it won't make any difference. And do we agree that the fetcher is loading all data for one type of equipment anyway ? It doesn't load just the data you need in the column ? So what you did just postpone the moment we're fetching the related equipments data of the new node (when you use it in a column) but it does not make a great difference if you fetch them node/alias declaration directly.

Reply:
But yes we have to think about this conception considering what will be the final usage of the spreadsheet. All columns will be customizable, and we will be able to load several nodes of the network for one given equipment type. So we have to think about it like 2 distinct systems, the node/alias system enables to fill more available data in our spreadhseet, and the custom column system that enables the users to combine nodes data as they want.